src/gt4sd/configuration.py

```diff
@@ -142,17 +171,23 @@ def sync_algorithm_with_s3(prefix: Optional[str] = None) -> str:
     return os.path.join(folder_path, prefix) if prefix is not None else folder_path
 
 
-def get_cached_algorithm_path(prefix: Optional[str] = None) -> str:
+def get_cached_algorithm_path(
+    prefix: Optional[str] = None, module: str = "algorithms"
+) -> str:
+
+    if module not in gt4sd_configuration_instance.gt4sd_s3_modules:
+        raise KeyError(f"Unknown submodule {module}")
```

Comment:
Here I would raise a `ValueError` since we explicitly check for the validity of the value. A final user does not need to know this is a dictionary key.
Also I would consider expanding the error message:
```suggestion
        raise ValueError(f"Unknown cache module: {module}. Supported module: {','.join(gt4sd_configuration_instance.gt4sd_s3_modules)}")
```